src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -276,9 +284,25 @@ private static string GenerateUniqueChannelLocalName()
 
                 lock (_sessionGuard)
```

Comment:
The lock is guarding reads and writes of _lastTask, and all execution (and script state and options) is within the tasks chained to _lastTask that execute outside of the lock. I've renamed _sessionGuard to _lastTaskGuard.